sklearn/decomposition/tests/test_nmf.py

```diff
@@ -932,3 +932,50 @@ def test_NMF_inverse_transform_W_deprecation():
 
     with pytest.warns(FutureWarning, match="Input argument `W` was renamed to `Xt`"):
         est.inverse_transform(W=Xt)
+
+
+@pytest.mark.parametrize("Estimator", [NMF, MiniBatchNMF])
+def test_nmf_n_components_auto(Estimator):
+    rng = np.random.RandomState(0)
```

Comment:
Please add a small description of the goal of this test, like "Check that n_components is correctly inferred from the provided custom initialization."

And similar for other tests

Reply:
done in https://github.com/scikit-learn/scikit-learn/pull/26634/commits/0e65acd8ff2ae46eb11dcfa5f5de5beb672302b2